build.sbt

```diff
@@ -1,73 +1,24 @@
-import sbtcrossproject.{crossProject, CrossType}
-
 val Scala212 = "2.12.15"
 val Scala213 = "2.13.8"
+val Scala3Version = "3.1.0"
 
-ThisBuild / crossScalaVersions := Seq(Scala213, Scala212, "3.1.0")
-ThisBuild / scalaVersion := Scala213
-
-ThisBuild / githubWorkflowJavaVersions := Seq("adopt@1.11")
-
-ThisBuild / githubWorkflowBuildMatrixAdditions += "platform" -> List("jvm", "js", "native")
-
-ThisBuild / githubWorkflowBuildMatrixExclusions ++=
-  (ThisBuild / crossScalaVersions).value.filter(_.startsWith("3.")).map { dottyVersion =>
-    MatrixExclude(Map("platform" -> "native", "scala" -> dottyVersion))
-  }
-
-val JvmCond = s"matrix.platform == 'jvm'"
-val JsCond = s"matrix.platform == 'js'"
-val NativeCond = s"matrix.platform == 'native'"
-
-val Scala212Cond = s"matrix.scala == '$Scala212'"
+ThisBuild / tlBaseVersion := "0.4"
 
-ThisBuild / githubWorkflowBuild := Seq(
-  WorkflowStep.Sbt(List("js/checkCI"), name = Some("Validate JavaScript"), cond = Some(JsCond)),
-  WorkflowStep.Sbt(List("native/checkCI"), name = Some("Validate Native"), cond = Some(NativeCond)),
-  WorkflowStep.Sbt(List("jvm/checkCI"),
-                   name = Some("Validate JVM"),
-                   cond = Some(JvmCond + " && " + s"matrix.scala != '$Scala212'")
-  ),
-  WorkflowStep.Use(UseRef.Public("actions", "setup-python", "v2"),
-                   name = Some("Setup Python"),
-                   params = Map("python-version" -> "3.x"),
-                   cond = Some(JvmCond + " && " + Scala212Cond)
-  ),
-  WorkflowStep.Run(List("pip install codecov"),
-                   name = Some("Setup codecov"),
-                   cond = Some(JvmCond + " && " + Scala212Cond)
-  ),
-  WorkflowStep.Sbt(List("coverage", "jvm/checkCI", "docs/mdoc", "coverageReport"),
-                   name = Some("Validate JVM (scala 2)"),
-                   cond = Some(JvmCond + " && " + Scala212Cond)
-  ),
-  WorkflowStep.Run(List("codecov"),
-                   name = Some("Upload Codecov Results"),
-                   cond = Some(JvmCond + " && " + Scala212Cond)
-  ),
-  WorkflowStep.Sbt(List("mimaReportBinaryIssues"),
-                   name = Some("Binary compatibility ${{ matrix.scala }}"),
-                   cond = Some(JvmCond + " && " + Scala212Cond)
-  )
-)
-
-ThisBuild / githubWorkflowAddedJobs ++= Seq(
-  WorkflowJob(
-    "checks",
-    "Format Scala code",
-    githubWorkflowJobSetup.value.toList ::: List(
-      WorkflowStep.Sbt(List("scalafmtCheckAll"), cond = Some(JvmCond + " && " + Scala212Cond)),
-      WorkflowStep.Sbt(List("scalafmtSbtCheck"), cond = Some(JvmCond + " && " + Scala212Cond))
-    ),
-    scalas = crossScalaVersions.value.toList
-  )
+ThisBuild / scalaVersion := Scala213
+ThisBuild / tlVersionIntroduced := Map("3" -> "0.4.2")
+ThisBuild / crossScalaVersions := Seq(Scala213, Scala212, Scala3Version)
+ThisBuild / githubWorkflowBuildMatrixExclusions +=
+  MatrixExclude(Map("project" -> "rootNative", "scala" -> Scala3Version))
```

Comment:
We'll need this to get snapshots since the primary branch is `master` not `main`.
```scala
ThisBuild / tlCiReleaseBranches := Seq("master")
```